src/scripts/main.js

```diff
@@ -1 +1,22 @@
 'use strict';
+
+document.addEventListener('DOMContentLoaded', function () {
+  const largeImg = document.getElementById('largeImg');
+  const thumbsList = document.getElementById('thumbs');
+
+  thumbsList.addEventListener('click', function (e) {
+    const clickedElement = e.target;
+
+    if (clickedElement.tagName === 'IMG' || clickedElement.tagName === 'A') {
+      e.preventDefault();
+
+      const newSrc =
+        clickedElement.tagName === 'IMG'
+          ? clickedElement.parentNode.href
+          : clickedElement.href;
```

Comment:
The logic here assumes that if the clicked element is an `IMG`, it is wrapped in an `A` tag, and the `href` of the parent node is used. This might not be true in all cases. Consider adding a check to ensure that `clickedElement.parentNode` is indeed an `A` tag before accessing `href` to avoid potential errors.